lib/puppet/resource_api/transport.rb

```diff
@@ -28,13 +28,27 @@ def list
   module_function :list # rubocop:disable Style/AccessModifierDeclarations
 
   def connect(name, connection_info)
-    validate(name, connection_info)
+    # clone the connection_info parameter, as validate may modify the value being passed in
```

Comment:
@DavidS can you take a look at this, the intention is to address the issue that `validate` may modify the connection_info hash.

Reply:
fixing the validation to not modify the `connection_info` would reduce GC pressure and compute overhead.

Reply:
Removed the dup and updated the way validation works. Running tests it seems that leaving unknown keys/values in place has had no negative effects.

Reply:
:crossed_fingers: